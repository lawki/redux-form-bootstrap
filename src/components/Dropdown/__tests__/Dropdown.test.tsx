import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { Dropdown } from '..';
import { DropdownOption } from "..";

export const options: DropdownOption[] = [
    {value: "You haven't selected a name yet!", type: "disabled"},
    {value: "Karen", type: "header"},
    {type: "divider"},
    {value: "Gillian", type: "header"},
    {type: "divider"},
    {value: "Bucky", type: "header"},
    {type: "divider"},
    {value: "Bikrum", type: "header"},
    {type: "divider"}
]


describe('Dropdown Component', () => {

    it('renders', () => {
        const fn = jest.fn();
        const app = shallow(
            <Dropdown
                text="Select a name"
                options={options}
                onChange={fn}
            />);
        expect(app).toMatchSnapshot();
    });

    it('has a single .btn-group', () => {
        const fn = jest.fn();
        const app = mount(
            <Dropdown
                text="Select a name"
                options={options}
                onChange={fn}
            />);
        expect(app.find('.btn-group').length).toBe(1);
    });

    it('handles onChange', () => {
        const fn = jest.fn();
        const app = mount(
            <Dropdown
                text="Select a name"
                options={options}
                onChange={fn}
            />);
        app.find('.btn-group').simulate('click');
        expect(fn.call.length).toBe(1);
    });
});