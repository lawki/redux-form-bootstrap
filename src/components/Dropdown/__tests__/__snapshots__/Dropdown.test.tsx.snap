// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component renders 1`] = `
<ButtonDropdown
  direction="down"
  isOpen={false}
  toggle={[Function]}
>
  <DropdownToggle
    aria-haspopup={true}
    caret={true}
    color="info"
    size="sm"
  >
    Select a name
  </DropdownToggle>
  <DropdownMenu
    flip={true}
    tag="div"
  >
    <DropdownItem
      disabled={true}
      divider={false}
      header={false}
      key="You haven't selected a name yet!_0"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
      You haven't selected a name yet!
    </DropdownItem>
    <DropdownItem
      disabled={false}
      divider={false}
      header={true}
      key="Karen_1"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
      Karen
    </DropdownItem>
    <DropdownItem
      disabled={false}
      divider={true}
      header={false}
      key="DropdownItem_2"
      onClick={[Function]}
      tag="button"
      toggle={true}
    />
    <DropdownItem
      disabled={false}
      divider={false}
      header={true}
      key="Gillian_3"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
      Gillian
    </DropdownItem>
    <DropdownItem
      disabled={false}
      divider={true}
      header={false}
      key="DropdownItem_4"
      onClick={[Function]}
      tag="button"
      toggle={true}
    />
    <DropdownItem
      disabled={false}
      divider={false}
      header={true}
      key="Bucky_5"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
      Bucky
    </DropdownItem>
    <DropdownItem
      disabled={false}
      divider={true}
      header={false}
      key="DropdownItem_6"
      onClick={[Function]}
      tag="button"
      toggle={true}
    />
    <DropdownItem
      disabled={false}
      divider={false}
      header={true}
      key="Bikrum_7"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
      Bikrum
    </DropdownItem>
    <DropdownItem
      disabled={false}
      divider={true}
      header={false}
      key="DropdownItem_8"
      onClick={[Function]}
      tag="button"
      toggle={true}
    />
  </DropdownMenu>
</ButtonDropdown>
`;
